pkg/kademlia/dialer.go

```diff
@@ -58,11 +62,16 @@ func (dialer *Dialer) Lookup(ctx context.Context, self pb.Node, ask pb.Node, fin
 		return nil, err
 	}
 
+	durationSeconds := &duration.Duration{
+		Seconds: int64(dialer.queryTimeout.Seconds()),
+	}
+
 	resp, err := conn.client.Query(ctx, &pb.QueryRequest{
-		Limit:    20, // TODO: should not be hardcoded, but instead kademlia k value, routing table depth, etc
-		Sender:   &self,
-		Target:   &find,
-		Pingback: true, // should only be true during bucket refreshing
+		Limit:        20, // TODO: should not be hardcoded, but instead kademlia k value, routing table depth, etc
+		Sender:       &self,
+		Target:       &find,
+		Pingback:     true, // should only be true during bucket refreshing
+		QueryTimeout: durationSeconds,
```

Comment:
Why is this a protobuf request param?

Reply:
I was thinking it could be sent from the client (inspector) but I could remove it.

Reply:
So the problem is that anyone can call this method with any value... however we want the actual server to decide the value. Not the remote untrusted caller.

Reply:
okay that makes sense